packages/reassure-compare/src/utils/readFileAndParseDataToFormat.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Util to read file content from path and filter file elements based on a JSON key
+ */
+
+import fs from 'fs/promises';
+
+export async function readFileAndParseDataToFormat(path: string, keyToFilterWith: string) {
```

Comment:
I think this function got a bit too generic with the `keyToFilterWith` prop. I would be happy to trade abstraction here to a small amount of `if` statements in the original `loadFile` function